src/Test.FunctionalTests.Sarif/Partitioning/FilterByPredicateTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Xunit.Abstractions;
+using Newtonsoft.Json;
+using Microsoft.CodeAnalysis.Test.Utilities.Sarif;
+using System;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+
+namespace Microsoft.CodeAnalysis.Sarif.FunctionalTests.Partitioning
+{
+    public class FilterByPredicateTests : FileDiffingFunctionalTests
+    {
+        public FilterByPredicateTests(ITestOutputHelper outputHelper, bool testProducesSarifCurrentVersion = true) :
```

Comment:
This might actually be a good case to not use the file diffing approach.
It seems like the verification criteria would be simple - how many Results left, did they all match the criteria, did only a few Artifacts come through.

A sample file for the inputs (to allow a lot of detail in them) makes sense, but since the code to verify them should be very short, it might be nicer to have that code then to introduce another dozen or two examples which might have to be updated if we change the format or output behavior in some way later.

I'm thinking selfishly about that 200+ file update when I changed SarifLogger. =)

Reply:
I understand your concern (which Michael shares) about having many granular test files. OTOH I like the file diffing approach because it's comprehensive.

Take a look at my comment in the test data file. I do _not_ intend to have (for example) one test file for artifacts, one for logical locations, one for code flows... Instead, I'll have one comprehensive test file. I think that addresses your concern.

---
In reply to: [338203971](https://github.com/microsoft/sarif-sdk/pull/1708#discussion_r338203971) [](ancestors = 338203971)